tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_ops.py

```diff
@@ -490,20 +512,41 @@ def embedding_lookup(
       def initial_value():
         return array_ops.zeros(initial_shape, dtype=params.value_dtype)
 
-    with ops.colocate_with(None, ignore_existing=True):
-      collections = [ops.GraphKeys.LOCAL_VARIABLES]
-      if params.trainable:
-        collections += [ops.GraphKeys.TRAINABLE_VARIABLES]
-      trainable_ = de.TrainableWrapper(params,
-                                       ids,
-                                       max_norm=max_norm,
-                                       initial_value=initial_value,
-                                       dtype=params.value_dtype,
-                                       trainable=params.trainable,
-                                       collections=collections,
-                                       model_mode=ModelMode.CURRENT_SETTING)
-      embeddings = array_ops.identity(trainable_)
-      embeddings = array_ops.reshape(embeddings, shape=embeddings_shape)
+  with ops.colocate_with(None, ignore_existing=True):
+    collections = [ops.GraphKeys.LOCAL_VARIABLES]
+    if params.trainable:
+      collections += [ops.GraphKeys.TRAINABLE_VARIABLES]
+
+    # TODO(Lifann) Use more resonable name.
+    def _create_trainable(trainable_name):
+      return de.TrainableWrapper(params,
```

Comment:
Plz check here , there is assume that 'TrainableWrapper' is always appear in the TW name, and the placer won't place TW onto PSs for multi-thread safe . [device placement](https://github.com/tensorflow/recommenders-addons/pull/119/files#diff-aa24a4847af774479cc7594ace62daeecf72558dc36dc8af02d1a1cdbbb3e079R204)

Reply:
Accept. Use `ops.colocate_with` to force the TW to reside with ids.